File: src/Web/vue-app/src/types/responses/succeededOrNotResponse.ts
import i18n from "@/i18n";
import '@/extensions/string.extensions';
import {Error} from "./error";

export class SucceededOrNotResponse {
    errors: Error[] = [];
    succeeded: boolean;

    constructor(succeeded: boolean, errors?: Error[]) {
        this.succeeded = succeeded;
        if (errors)
            this.errors = errors
    }

    getErrorMessages(translationLocation: string, fallbackKey?: string) {
        const errorMessages: string[] = [];
        this.errors.forEach(error => {
            const errorKey = error.errorType.makeFirstLetterLowercase();
            const errorMessage = i18n.t(`${translationLocation}.${errorKey}`);
            if (fallbackKey)
                errorMessages.push(errorMessage ? errorMessage : i18n.t(fallbackKey));
            else
                errorMessages.push(errorMessage ? errorMessage : i18n.t('validation.errorOccured'))
        });
        return errorMessages
    }
}

File: src/Web/Features/Common/SucceededOrNotResponse.cs
using Application.Common;

namespace Web.Features.Common;

public class SucceededOrNotResponse
{
    public SucceededOrNotResponse() { }

    public SucceededOrNotResponse(bool succeeded, Error? error = null)
    {
        Succeeded = succeeded;
        if (error != null)
        {
            Errors.Add(error);
        }
    }

    public SucceededOrNotResponse(bool succeeded, IEnumerable<Error> errors)
    {
        Succeeded = succeeded;
        Errors.AddRange(errors);
    }

    public bool Succeeded { get; set; }
    public List<Error> Errors { get; set; } = [];
}